AirCasting/CreateSessionViews/ConfirmCreatingSessionView.swift

```diff
@@ -20,16 +20,32 @@ struct ConfirmCreatingSessionView: View {
     @State private var isPresentingAlert: Bool = false
     @EnvironmentObject var selectedSection: SelectSection
     @EnvironmentObject private var sessionContext: CreateSessionContext
-    @InjectedObject private var locationTracker: LocationTracker
+    @Injected private var locationTracker: LocationTracker
     @EnvironmentObject private var tabSelection: TabBarSelection
     @Binding var creatingSessionFlowContinues: Bool
 
     var sessionName: String
     private var sessionType: String { (sessionContext.sessionType ?? .fixed).description.lowercased() }
+    
+    init(creatingSessionFlowContinues: Binding<Bool>, sessionName: String) {
+        _creatingSessionFlowContinues = .init(projectedValue: creatingSessionFlowContinues)
+        self.sessionName = sessionName
+    }
 
     var body: some View {
         LoadingView(isShowing: $isActive) {
             contentViewWithAlert
+                .onAppear {
+                    if sessionContext.sessionType == .mobile && sessionContext.locationless != true {
+                        // We need to start tracking location to save the most recent location as the session starting location
+                        locationTracker.start()
+                    }
+                }
+                .onDisappear {
+                    if sessionContext.sessionType == .mobile && sessionContext.locationless != true {
```

Comment:
`sessionContext.sessionType == .mobile && sessionContext.locationless != true`

can be extracted to a `let` as it's used twice